domains/runtime/evm/src/lib.rs

```diff
@@ -856,11 +860,34 @@ impl_runtime_apis! {
         }
 
         fn check_transaction_validity(
-            _uxt: &<Block as BlockT>::Extrinsic,
-            _block_hash: <Block as BlockT>::Hash,
+            uxt: &<Block as BlockT>::Extrinsic,
+            block_hash: <Block as BlockT>::Hash,
         ) -> Result<(), domain_runtime_primitives::CheckTxValidityError> {
-            // TODO: check transaction fee to core-evm
-            Ok(())
+            let lookup = frame_system::ChainContext::<Runtime>::default();
+            let maybe_signer_info = extract_signer_inner(uxt, &lookup);
+
+            if let Some(signer_info) = maybe_signer_info {
+                let signer = signer_info.map_err(|tx_validity_error| {
+                    domain_runtime_primitives::CheckTxValidityError::UnableToExtractSigner {
+                        error: tx_validity_error
+                    }
+                })?;
+                let tx_validity =
+                    Executive::validate_transaction(TransactionSource::External, uxt.clone(), block_hash);
+
+                tx_validity.map(|_| ()).map_err(|tx_validity_error| {
+                    let storage_keys = sp_std::vec![
```

Comment:
I might have some missing context here but are you sure these are the storage keys that are accessed to validate an transaction ?

Reply:
From my understanding yes. But, I am going through the code path once again. Once done, I will write test as well as update the keys here, if I have missed any.

Reply:
This is also the storage keys used in the last invalid transaction fraud proof impl 
https://github.com/subspace/subspace/blob/3fb43667110c90814e5231cedc77a0f3cd6afd31/domains/runtime/system/src/runtime.rs#L582-L590

Reply:
@NingLin-P No there are more keys used in the SignedExtension. It looks like in previous iteration we missed them. I am writing test atm and once finished will be pushing it here.